examples/tomo/data/raw_binary_io_with_header.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Example for the usage of raw binary reader/writer with header.
+
+The code below defines a simple file format for 2D images using
+    - shape
+    - origin
+    - pixel size
+
+First, a file is written using `FileWriterRawBinaryWithHeader`. This
+requires a header in a certain format.
+Then, the same file is read again using a file specification and the
+`FileReaderRawBinaryWithHeader`. The specification is given as a
+sequence of dictionaries with a certain structure.
+"""
+
+from __future__ import print_function
+
+from collections import OrderedDict
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy
+import tempfile
+
+from odl.tomo.data import (
+    FileReaderRawBinaryWithHeader, FileWriterRawBinaryWithHeader)
+
+
+# --- Writing --- #
+
+# Create some test data. We arbitrarily define origin and pixel size.
+# In practice, these could come from a `DiscreteLp` space as `mid_pt`
+# and `cell_sides` properties.
+image = scipy.misc.ascent()
+shape = np.array(image.shape, dtype='int32')
+origin = np.array([-1.0, 0.0], dtype='float32')
+px_size = np.array([0.1, 0.1], dtype='float32')
+# To make it storable as binary data, we take the string version of the data
+# type with a fixed size of 10 characters and encode it as array of single
+# bytes.
+dtype = np.fromiter(str(image.dtype).ljust(10), dtype='S1')
+
+# Create the header
+# Use an OrderedDict for the header to have a predictable order when
```

Comment:
It is -- otherwise the header is stored in random order in the file. The header writing routine loops over `header.items()` and that order is unpredictable for regular `dict`s.